### DISCUSSION TITLE
Next js ssr deployment

### DISCUSSION THREAD

**Discussion by shahebazshaikh99** [vercel/next.js#54771] [Help]
### Summary

i want to deploy my next js app which included with ssr and gsp when i run a command npm run build then the .next file generating with the size of 1.9gb.

so how can i optimized it to lower??

**Comment by hyjoong** [comment]
1. Bundle Size Analysis
Use tools like @next/bundle-analyzer to analyze your bundle size. This will help you identify which parts of your code are contributing the most to the bundle size.

Reference Links:
- [npm @next/bundle-analyzer package](https://www.npmjs.com/package/@next/bundle-analyzer)
- [Analyzing Next.js App Bundles](https://blog.logrocket.com/how-analyze-next-js-app-bundles/)

2. App Router Migration
By using the App Router in Next.js 13, you can write code in Server Components to fetch data instead of using SSR. This approach minimizes the impact on the client bundle size.

Optimize your Next.js SSR deployment through bundle size analysis and migrating your code to the App Router.

Reference Links:
- https://nextjs.org/docs/app/building-your-application/rendering/server-components#benefits-of-server-rendering
